phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java

```diff
@@ -337,6 +384,22 @@ public boolean nextRaw(List<Cell> result) throws IOException {
                         arrayElementCell = result.get(arrayElementCellPosition);
                     }
                     if (ScanUtil.isLocalIndex(scan) && !ScanUtil.isAnalyzeTable(scan)) {
```

Comment:
We'll need some code comments here. Are we skipping until we find the actualStartKey row? Would it be better/possible to do a seek? This seems like it'd be potentially expensive, no?